wallet/src/actors/app/methods.rs

```diff
@@ -335,13 +335,15 @@ impl App {
                 slf.state
                     .create_session(session_id.clone(), wallet_id.clone(), wallet.clone());
 
-                // Start synchronization for this wallet
-                let sink = slf.state.get_sink(&session_id);
-                slf.params.worker.do_send(worker::SyncRequest {
-                    wallet_id,
-                    wallet,
-                    sink,
-                });
+                // If the node is synced start synchronization for this wallet
+                if slf.state.node_state == Some(StateMachine::Synced) {
+                    let sink = slf.state.get_sink(&session_id);
+                    slf.params.worker.do_send(worker::SyncRequest {
+                        wallet_id,
+                        wallet,
+                        sink,
+                    });
+                };
```

Comment:
What happens if the node is not synced? I guess the synchronization will start later, but when?

Reply:
It will start when the wallet receives the synced notification from the node